tensorboard/plugins/audio/summary_v2.py

```diff
@@ -103,15 +107,31 @@ def lazy_tensor():
             tf.debugging.assert_rank(data, 3)
             tf.debugging.assert_non_negative(max_outputs)
             limited_audio = data[:max_outputs]
+
             encode_fn = functools.partial(
                 audio_ops.encode_wav, sample_rate=sample_rate
             )
-            encoded_audio = tf.map_fn(
-                encode_fn,
-                limited_audio,
-                dtype=tf.string,
-                name="encode_each_audio",
-            )
+            if lengths is not None:
+                tf.debugging.assert_rank(lengths, 1)
+                limited_lengths = lengths[:max_outputs]
+
+                def encode_with_length(datum_length):
```

Comment:
nit: Rename to `datum_and_length`? I was confused on first read, parsing
this as “length of datum”.

Reply:
Done.